Composer/packages/client/src/telemetry/TelemetryClient.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { LogData, TelemetryEventName, TelemetryEvents, TelemetrySettings } from '@bfc/shared';
+
+import AppInsightsClient from './AppInsightsClient';
+import ConsoleClient from './ConsoleClient';
+
+export default class TelemetryClient {
+  private static _getProperties?: () => LogData;
+  private static _telemetrySettings?: TelemetrySettings;
+
+  public static setup(telemetrySettings: TelemetrySettings, additionalProperties: LogData | (() => LogData)) {
+    if (this._telemetrySettings?.allowDataCollection !== telemetrySettings.allowDataCollection) {
+      this.client?.drain();
+    }
+
+    this._getProperties =
+      typeof additionalProperties === 'function' ? additionalProperties : () => additionalProperties;
```

Comment:
What I would do instead is store `additionalProperties` as a private property and define a getter like this:

```ts
private static additionalProperties?: () => LogData;

private static get sharedProperties() {
  return {
    ...this.additionalProperties?.(),
    timestamp: Date.now(),
    composerVersion: process.env.COMPOSER_VERSION || 'unknown',
    sdkPackageVersion: process.env.SDK_PACKAGE_VERSION || 'unknown',
  }
}
```